packages/@aws-cdk/aws-ivs/lib/channel.ts

```diff
@@ -0,0 +1,161 @@
+import * as core from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnChannel } from './ivs.generated';
+import { StreamKey } from './stream-key';
+
+/**
+ * Represents an IVS Channel
+ */
+export interface IChannel extends core.IResource {
+  /**
+   * The channel ARN. For example: arn:aws:ivs:us-west-2:123456789012:channel/abcdABCDefgh
+   *
+   * @attribute
+   */
+  readonly channelArn: string;
+
+  /**
+   * Adds a stream key for this IVS Channel.
+   * @param id construct ID
+   */
+  addStreamKey(id: string): StreamKey;
+}
+
+/**
+ * Reference to a new or existing IVS Channel
+ */
+abstract class ChannelBase extends core.Resource implements IChannel {
+  public abstract readonly channelArn: string;
+
+  public addStreamKey(id: string): StreamKey {
+    return new StreamKey(this, id, {
+      channel: this,
+    });
+  }
+}
+
+/**
+  Channel latency mode
+*/
+export enum LatencyMode {
+  /**
+   * Use LOW to minimize broadcaster-to-viewer latency for interactive broadcasts.
+   */
+  LOW = 'LOW',
+
+  /**
+   * Use NORMAL for broadcasts that do not require viewer interaction.
+   */
+  NORMAL = 'NORMAL',
+}
+
+/**
+  * The channel type, which determines the allowable resolution and bitrate.
+  * If you exceed the allowable resolution or bitrate, the stream probably will disconnect immediately.
+*/
+export enum ChannelType {
+  /**
+   * Multiple qualities are generated from the original input, to automatically give viewers the best experience for
+   * their devices and network conditions.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ivs-channel.html
+   */
+  STANDARD = 'STANDARD',
+
+  /**
+   * delivers the original input to viewers. The viewer’s video-quality choice is limited to the original input.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ivs-channel.html
+   */
+  BASIC = 'BASIC',
+}
+
+/**
+ * Properties for creating a new Channel
+ */
+export interface ChannelProps {
+  /**
+   * Whether the channel is authorized.
+   *
+   * If you wish to make an authorized channel, you will need to ensure that
+   * a PlaybackKeyPair has been uploaded to your account as this is used to
+   * validate the signed JWT that is required for authorization
+   *
+   * @default false
+   */
+  readonly authorized?: boolean;
+
+  /**
+   * Channel latency mode.
+   *
+   * @default LatencyMode.LOW
+   */
+  readonly latencyMode?: LatencyMode;
+
+  /**
+   * Channel name.
+   *
+   * @default - None
+   */
+  readonly name?: string;
```

Comment:
By convention, these are prefixed with the class name -- this is why you've got all the awslint exceptions in `package.json`. Renaming to `channelName` will remove the need for the exceptions.

Reply:
I thought the lint rule was specifically for resources with a physicalName. None of these resources require a name so I thought that meant that they weren't resources with a physicalName. I can change all these if it makes sense, it's just not really that obvious that I needed to do it.

Reply:
Is it that Channels (for example), don't need to have names, or just that names are auto-generated if one isn't provided? If the former, then I can agree with the linter exclusions and this pattern; if the latter, we should treat these as any other resource name. For example, S3 buckets don't "require" a name either (via CloudFormation), but one is generated.

Reply:
Names are not required and are not auto generated